### PR TITLE
Support Heroku

### PR DESCRIPTION
**Pull request by arreche**
See:
* https://devcenter.heroku.com/articles/procfile

### CODE REVIEW

**Comment by CasiOo** [comment]
What's the reasoning behind this PR?
Is it aiming to fix the 404, we were getting for bundle.js?
Are there any mongo setup, that were missing before?

File: app.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "scripts": {},
+  "env": {
+    "NPM_CONFIG_PRODUCTION": {
+      "required": true
+    }
+  },
+  "addons": [
+    "mongolab"
+  ]
+}
```
Comment (CasiOo) [inline app.js]: Should this have the .js extension and not .json ?
https://blog.heroku.com/archives/2014/5/22/introducing_the_app_json_application_manifest

**Comment by arreche** [comment]
This PR supports to deploy our application on the [Heroku](https://www.heroku.com/about) platform.

Heroku requires to specify how to run applications in a [Procfile](https://devcenter.heroku.com/articles/procfile#procfile-naming-and-location) file.

Our configuration runs a single web process to build the front-end and then starts the backend server. (The build part is a a temporal work around for the 404 that we are getting for bundle.js.)

The backend was also adapted to setup database URLs dynamically in order to support both local and [Heroku environments](https://devcenter.heroku.com/articles/mongolab#getting-your-connection-uri)

**Comment by CasiOo** [comment]
:+1: looks good

**Pull request merged**